modules/pricer/src/test/java/com/opengamma/strata/pricer/swaption/SabrSwaptionCalibratorSmileTestUtils.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.swaption;
+
+import static org.testng.Assert.assertTrue;
+
+import java.time.Period;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.collect.array.DoubleMatrix;
+import com.opengamma.strata.market.ValueType;
+import com.opengamma.strata.pricer.calibration.RawOptionData;
+
+/**
+ * Utilities for the different tests related to {@link SabrSwaptionCalibrator}
+ */
+public class SabrSwaptionCalibratorSmileTestUtils {
+
+  /**
+   * Create a {@link RawOptionData} object for calibration from data.
+   * 
+   * @param strikeLikeType  the type of the strike-like dimension
+   * @param strikeLikeData  the data related to the strik-like dimension
+   * @param expiries  the list of expiries
+   * @param dataType  the type of the data
+   * @param dataArray  the array with the raw data, including potential Double.NaN for missing data. 
+   * @return the raw option data object
+   */
+  public static List<RawOptionData> rawData(
+      ValueType strikeLikeType,
+      DoubleArray strikeLikeData,
+      List<Period> expiries,
+      ValueType dataType,
+      double[][][] dataArray) {
+    List<RawOptionData> raw = new ArrayList<>();
+    for (int looptenor = 0; looptenor < dataArray.length; looptenor++) {
+      raw.add(RawOptionData.of(strikeLikeData, strikeLikeType, expiries,
+          DoubleMatrix.ofUnsafe(dataArray[looptenor]), dataType));
+    }
+    return raw;
+  }
+
+  /**
+   * Create a {@link RawOptionData} object for calibration from data and shift one point.
+   * 
+   * @param strikeLikeType  the type of the strike-like dimension
+   * @param strikeLikeData  the data related to the strik-like dimension
+   * @param expiries  the list of expiries
+   * @param dataType  the type of the data
+   * @param dataArray  the array with the raw data, including potential Double.NaN for missing data. 
+   * @param i  the index of the tenor to shift
+   * @param j  the index of the expiry to shift
+   * @param k  the index of the strike-like dimension to shift
+   * @param shift  the size of the shift
+   * @return the raw option data object
+   */
+  public static List<RawOptionData> rawDataShiftPoint(
+      ValueType strikeLikeType,
+      DoubleArray strikeLikeData,
+      List<Period> expiries,
+      ValueType dataType,
+      double[][][] dataArray, int i, int j, int k, double shift) {
```

Comment:
Coding style has each parameter on a new line with a blank line as the first line of method